### PR TITLE
Code improvements/consistency

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change
Code improvements identified using [lgtm.org](https://lgtm.com/projects/g/hyperspy/rosettasciio?mode=tree&severity=&id=)

For "Modification of parameter with default" see explanation at https://lgtm.com/rules/4840097/

Otherwise, commit messages should be self explanatory. Mainly removes unused lines of code (unused imports, variables, or unreachable lines of code).

A few errors in the code turned up along the way.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [n/a] update docstring (if appropriate),
- [n/a] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/rosettasciio/blob/main/upcoming_changes/README.rst)),
- [ ] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [n/a] add tests,
- [x] ready for review.

### CODE REVIEW

File: rsciio/usid_hdf5/api.py [hunk 1]
```diff
@@ -432,10 +432,6 @@ def file_reader(
             h5_dset, ignore_non_uniform_dims=ignore_non_uniform_dims, lazy=lazy, **kwds
         )
 
-    # At least close the file handle if not lazy load
-    if not lazy:
```
Comment (ericpre) [inline rsciio/usid_hdf5/api.py:436]: When not loading lazily, we still need to close the file.

**Comment by ericpre** [comment]
I am surprised that these errors doesn't appear in https://github.com/hyperspy/rosettasciio/security/code-scanning?

File: rsciio/usid_hdf5/api.py (hunk 1 above)
Comment (jlaehne) [reply rsciio/usid_hdf5/api.py:436]: But these two lines are unreachable code:
https://lgtm.com/projects/g/hyperspy/rosettasciio?mode=list&id=py%2Funreachable-statement

The preceding `if/else` are both ending with a `return` statement, so the file closing would not be executed. So I guess, we have to close the file before the return statement.

Comment (jlaehne) [reply rsciio/usid_hdf5/api.py:436]: Implemented.

**Comment by jlaehne** [comment]
> I am surprised that these errors doesn't appear in https://github.com/hyperspy/rosettasciio/security/code-scanning?

I am indeed startled as well, because both should be using the same codeQL engine to scan the code.

File: rsciio/usid_hdf5/api.py (hunk 1 above)
Comment (jlaehne) [reply rsciio/usid_hdf5/api.py:436]: ok, done

Comment (ericpre) [reply rsciio/usid_hdf5/api.py:436]: To simplify the code, would it be better to keep only one return at the end of the function?

**Pull request merged**